generatebundlefile/ecr_public.go

```diff
@@ -207,12 +207,29 @@ func (c *ecrPublicClient) GetPublicAuthToken() (string, error) {
 }
 
 // getNameAndVersionPublic looks up the latest pushed helm chart's tag from a given repo name Full name in Public ECR OCI format.
-func (c *SDKClients) getNameAndVersionPublic(repoName, registryURI string) (string, string, string, error) {
+func (c *SDKClients) getNameAndVersionPublic(repoName, tag, registryURI string) (string, string, string, error) {
 	var version string
 	var sha string
 	splitname := strings.Split(repoName, ":") // TODO add a regex filter
 	name := splitname[0]
-	if len(splitname) == 1 {
+	ecrname := fmt.Sprintf("%s/%s", c.ecrPublicClient.SourceRegistry, name)
+	if len(splitname) > 0 {
+		if !strings.Contains(tag, "latest") {
+			var imagelookup []ecrpublictypes.ImageIdentifier
+			imagelookup = append(imagelookup, ecrpublictypes.ImageIdentifier{ImageTag: &tag})
+			ImageDetails, err := c.ecrPublicClient.DescribePublic(&ecrpublic.DescribeImagesInput{
+				RepositoryName: aws.String(repoName),
+				ImageIds:       imagelookup,
+			})
```

Comment:
```suggestion
		if !strings.Contains(tag, "latest") {
			imageIDs := []ecrpublictypes.ImageIdentifier{ecrpublictypes.ImageIdentifier{ImageTag: &tag}}
			ImageDetails, err := c.ecrPublicClient.DescribePublic(&ecrpublic.DescribeImagesInput{
				RepositoryName: aws.String(repoName),
				ImageIds:       imagelookup,
			})
```

This feels like it's repeated from earlier in the review. Does it make sense to extract it into a well-named helper function?